src/providers/environment/environment.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from 'react'
+
+import { EnvironmentState, EnvironmentStoreActions, useEnvironmentStore } from './store'
+
+type EnvironmentContextValue = EnvironmentStoreActions & EnvironmentState
+
+const EnvironmentContext = React.createContext<undefined | EnvironmentContextValue>(undefined)
+EnvironmentContext.displayName = 'EnvironmentContext'
+
+export const EnvironmentProvider: React.FC = ({ children }) => {
+  const { setSelectedNode, selectedNode, setTargetEnv, targetEnv } = useEnvironmentStore((state) => ({
+    ...state.actions,
+    ...state,
+  }))
+
+  const contextValue: EnvironmentContextValue = {
+    selectedNode,
+    setSelectedNode,
+    setTargetEnv,
+    targetEnv,
+  }
+
+  return <EnvironmentContext.Provider value={contextValue}>{children}</EnvironmentContext.Provider>
+}
```

Comment:
Why do we need this provider at all? I think all this info could be extracted directly from store?

Reply:
I did this because at first I thought we use it in several places. But you are right, it's better to get rid of this.